src/foam/nanos/bench/benchmarks/AuthorizerBenchmark.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.bench.benchmarks;
+
+import foam.core.X;
+import foam.dao.ArraySink;
+import foam.dao.DAO;
+import foam.dao.MDAO;
+import foam.nanos.auth.AuthorizationDAO;
+import foam.nanos.auth.Authorizer;
+import foam.nanos.auth.User;
+import foam.nanos.bench.Benchmark;
+import foam.nanos.dao.Operation;
+import foam.test.TestUtils;
+
+/**
+  General benchmark authorizer
+  {@authorizer} wraps target DAO using authorizer provided
+  {@recordAmount} defines amount of records to be created for benchmark (creates users in new MDAO)
+  {@op} a (CRUD) operation to run benchmark on.
+  Each CRUD operation has its own method specified for benchmarking, meaning each benchmark will relate to a single operation.
+  If further setup or additional logic on each CRUD operation is required, extending this class and referencing the setup and CRUD method should suffice.
+ */
+
+public class AuthorizerBenchmark
+  implements Benchmark
+{
+  protected X userAuthorizedContext;
+  protected DAO dao = new MDAO(User.getOwnClassInfo());
+  protected Authorizer authorizer;
+  protected int recordAmount;
```

Comment:
Believe this is the same as the BenchmarkRunner invocationCount. 
When BenchmarkRunner is configured for multiple threads, then it executes invocationCount per thread.

Reply:
I thought so. I made recordAmount so this benchmark is capable of benchmarking each thread with multiple records instead of one entry or a hard coded iteration per thread.